evm_trace/gas.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List
+
+from evm_trace.base import CallTreeNode
+
+Address = str
+Calldata = bytes
+GasUsage = int
+
+GasReport = Dict[Address, Dict[Calldata, List[GasUsage]]]
+
+
+def extract_gas_report(calltree: CallTreeNode) -> GasReport:
```

Comment:
I tend to find that anytime `extract_` is in a method name, `get_` works just as good and is shorter and more consistent. Sure, it is an "extraction" from the call tree, but also we can "get" it from the call tree